src/sinks/util/tcp.rs

```diff
@@ -79,10 +86,11 @@ enum TcpSinkState {
     ResolvingDns(crate::dns::ResolverFuture),
     Connecting(MaybeTlsConnector),
     Connected(TcpOrTlsStream),
-    Backoff(Delay),
+    Backoff(Box<dyn Future<Item = (), Error = ()> + Send>),
```

Comment:
I wan wondering how difficult would it be to convert this part to `futures` `0.3` and user actual `Delay` here? Not that it's required for this PR, but if it's easy it would be nice to have as well.

Reply:
I think should be pretty simple when we move `trait Sink` to futures:0.3 :thinking:

Reply:
Ah, we probably don't want to hold this off until that arrives... :+1: